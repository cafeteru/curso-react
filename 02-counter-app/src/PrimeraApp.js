import React from "react";

const PrimeraApp = ({ saludo = "Eyyyyyyy" }) => {
  return (
    <>
      <h1>{saludo}</h1>
      <p>Mi primera aplicaci√≥n</p>
    </>
  );
};

export default PrimeraApp;
