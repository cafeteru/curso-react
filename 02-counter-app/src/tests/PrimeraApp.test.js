import PrimeraApp from "../PrimeraApp";
import { shallow } from "enzyme";

describe("PrimeraApp", () => {
  const saludo = "Hola mundo";

  test("Debe de mostrar el <PrimeraApp /> correctamente", () => {
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot(); //
  });

  test("Debe de mostrar el subtitulo", () => {
    const subtitulo = "Soy Iv√°n";
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
    );
    const textParrafo = wrapper.find("p").text();
    expect(textParrafo).toBe(subtitulo);
  });
});
