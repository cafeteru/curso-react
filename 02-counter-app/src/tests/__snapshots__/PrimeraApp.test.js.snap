// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeraApp Debe de mostrar el <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola mundo
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
